### PR TITLE
Implement improvements in APT keys management

### PR DESCRIPTION
**Pull request by bkabrda**
* Get keys from keys.datadoghq.com
* Add 'signed-by' option to all sources list lines, point them to the dynamically constructed keyring

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -86,23 +86,17 @@ fi
...
-
-if [[ $STACK == "heroku-16" ]]; then
-  gpg --ignore-time-conflict --no-options --no-default-keyring --homedir "$GPG_HOME_DIR" --keyring "$APT_KEYRING" --import "$GPG_KEY_FILE" | indent
-else
-  gpg --ignore-time-conflict --no-options --no-default-keyring --homedir "$GPG_HOME_DIR" --keyring "$APT_KEYBOX" --trustdb "$APT_KEYRING" --import "$GPG_KEY_FILE" | indent
-  gpg --no-default-keyring --keyring "$APT_KEYBOX" --export > $APT_KEYRING
-fi
+for key in "${DATADOG_APT_KEYS[@]}"; do
+  curl -o "/tmp/${key}" "${DATADOG_APT_KEYS_URL}/${key}" | indent
+  gpg --ignore-time-conflict --no-options --no-default-keyring --homedir "$GPG_HOME_DIR" --keyring "$DATADOG_APT_KEYRING" --import "/tmp/${key}" | indent
```
Comment (albertvaka) [inline bin/compile:98]: What are those `| indent` doing everywhere? 😂

File: test/compile_and_run_test.sh [hunk 2]
```diff
@@ -59,7 +59,9 @@ compileAndRunVersion()
 
 testReleased6Versions()
 {
+  echo "deb [trusted=yes] https://apt.datadoghq.com/ stable 6" > "${BUILDPACK_HOME}/etc/datadog.list"
```
Comment (albertvaka) [inline test/compile_and_run_test.sh:62]: Why adding a line with `trusted=yes` then replacing it with `signed-by`?

**Review by albertvaka — APPROVED** [review]
LGTM, but I'm no expert in Heroku. Left some questions for things I don't fully understand :)

File: bin/compile (hunk 1 above)
Comment (bkabrda) [reply bin/compile:98]: These are used throughout the script to indent all commands that have some output to make it look prettier + consistent when running the buildpack.

File: test/compile_and_run_test.sh (hunk 2 above)
Comment (bkabrda) [reply test/compile_and_run_test.sh:62]: The test first gets available versions, before running the `bin/compile` script. This means that at this point the GPG keyring is not created yet and so we have to skip repodata verification at this point. After we get the list of versions, we revert the source list file to the previous state and then run `bin/compile` (by running `compileAndRunVersion`), which does its own thing, now while actually using `signed-by`.

File: bin/compile (hunk 1 above)
Comment (albertvaka) [reply bin/compile:98]: Didn't know the command 👌

Comment (bkabrda) [reply bin/compile:98]: FWIW it's not a command, it's a function implemented in `bin/common.sh`

**Review by arapulido — APPROVED** [review]
LGTM

**Pull request merged**